doc/CONTRIBUTING.md

```diff
@@ -1,59 +1,107 @@
+- Save your changes by creating own local branches on git
 
-This project uses the [flake8](http://flake8.pycqa.org/en/latest) method of
-code style enforcement, linting, and checking.
+## Add, Commit and Push
 
-All python code contributed to pbench must match the style requirements. These
-requirements are enforced by the [pre-commit](https://pre-commit.com) hook
-using the [black](https://github.com/psf/black) Python code formatter.
+- Follow these commands to push the changes to your branch.
+```
+git add .
+git commit -m "Issue solved"
+git push origin branch_name
+```
 
-## Use pre-commit to set automatic commit requirements
+## Conventions on commits, PRs, and overall git best practices.
 
-This project makes use of [pre-commit](https://pre-commit.com/) to do automatic
-lint and style checking on every commit containing Python files.
+- Commit messages should have a short description (50 - 70 characters) followed by a longer format description of the changes below if needed. You’ll also notice each line is formatted for a specific length in the longer format description. For example:
+Extend auditing to incoming, results, and users
 
-To install the pre-commit hook, run the executable from your Python 3 framework
-while in your current pbench git checkout:
+```
+The server audit is now applied to the incoming, results, and users
+directory hierarchies.  Any unpacked tar ball should now be compre-
+hensively checked to see that all is in the correct place.
+
+The test-20 unit test gold file holds an example of an audit report
+covering all the possible outputs it can emit.  Each unit test runs
+the report as well, and they have been updated accordingly.
+```
 
+- For more on best practices, check out this article for reference from time to time: https://www.git-tower.com/learn/git/ebook/en/command-line/appendix/best-practices
+
+## Opening a pull request
+1. **If there are multiple commits, squash down the commits to one**
+  a. For more complicated commits it is appropriate to have more than one
+2. Commit the changes
+3. Click on New Pull Request
+4. Write appropriate Pull Request Title stating the fix
+  a. Use present tense (ex. Fixes, Changes, Fixing, Changing..)
+5. References the issue that the PR is fixing with **“Fixes #issue_number”** in the description
+6. Provide a detailed description (at least 50 - 70 characters) of the changes.(If UI, add screenshots or gif)
+7. Make sure that the branches can be automatically merged (otherwise rebase the PR with master) and then click : `Create pull request`
+![Image](https://lh5.googleusercontent.com/V14SjFhimKYF1fH6TXMfaoZtDCj2ZH0d9USqe8YHyn0xOOVYekiXtx2CwdOQbSvxWPB6JVEfi4jSM_mjkSMaaI7voQNYQ8gDWntMhzCMbj3wrK3H4eCSEsdVq_XP_aZMdb5h9xU4)
+
+8. Assign the PR to yourself and add appropriate labels
+9. Add "**WIP**" label if the work is still in progress
+10. Add "**DO NOT MERGE**" label if the work is not needed to be merged or there is no agreement on the work yet
```

Comment:
Draft PR?

Reply:
Updated this instruction to create a `Draft` PR first, and then mark it ready for review.